strawberryfields/engine.py

```diff
@@ -277,15 +271,35 @@ def _broadcast_nones(val, dim):
                 p = p.compile(target, **compile_options)
             p.lock()
 
-            self._run_program(p, **kwargs)
-            shots = kwargs.get("shots", 1)
-            self.samples = [_broadcast_nones(p.reg_refs[k].val, shots) for k in sorted(p.reg_refs)]
+            _, values = self._run_program(p, **kwargs)
```

Comment:
just for my own understanding: was it not necessary to have the `values` be returned because previously we also checked the `RegRef` instances and built up `self.samples` that way?

Reply:
Yes, that's correct. So, previously the samples were only built up from `RegRef` and then `None` was added to all the non-measured modes (with `_broadcast_nones`). No results were actually returned directly from the backends.